### PR TITLE
Stock Ageing TypeError: ‘<’ not supported between instances of ‘int’ …

### PR DESCRIPTION
**Pull request by clarkejj**
…and ‘str’

Re this user report https://discuss.erpnext.com/t/stock-ageing-report-typeerror-not-supported-between-instances-of-int-and-str/62605

Notes: 

ASSUMPTION:  This error arises in current version-12 production?

WARNING: This proposed change has not been tested

> Please provide enough information so that others can review your pull request:

> Explain the **details** for making this change. What existing problem does the pull request solve?

> Screenshots/GIFs

### CODE REVIEW

File: erpnext/stock/report/stock_ageing/stock_ageing.py [hunk 1]
```diff
@@ -180,14 +180,14 @@ def get_fifo_queue(filters, sle=None):
 				qty_to_pop = abs(d.actual_qty)
 				while qty_to_pop:
 					batch = fifo_queue[0] if fifo_queue else [0, None]
-					if 0 < batch[0] <= qty_to_pop:
+					if 0 < cint(batch[0]) <= qty_to_pop:
```
Comment (deepeshgarg007) [inline erpnext/stock/report/stock_ageing/stock_ageing.py]: `flt` should be used instead of `cint` as qty can be fraction value as well

**Review by deepeshgarg007 — CHANGES_REQUESTED** [review]

**Review by clarkejj — COMMENTED** [review]
As Deepesh advised on review of my original proposed commit - change cint to flt since batch[0] can be int or float - see line 49

Thanks Deepesh!

File: erpnext/stock/report/stock_ageing/stock_ageing.py [hunk 2]
```diff
@@ -180,14 +180,14 @@ def get_fifo_queue(filters, sle=None):
...
+					if 0 < flt(batch[0]) <= qty_to_pop:
 						# if batch qty > 0
 						# not enough or exactly same qty in current batch, clear batch
-						qty_to_pop -= batch[0]
+						qty_to_pop -= flt(batch[0])
 						transferred_item_details[(d.voucher_no, d.name)].append(fifo_queue.pop(0))
 					else:
 						# all from current batch
-						batch[0] -= qty_to_pop
+						flt(batch[0]) -= qty_to_pop
```
Comment (marination) [inline erpnext/stock/report/stock_ageing/stock_ageing.py]:
```suggestion
						batch[0] = flt(batch[0]) - qty_to_pop
```
Can't assign values to a function call so `flt(batch[0])` is problematic

**Review by clarkejj — COMMENTED** [review]
Many thanks for your patient pointer Marica - that assignment violation is clear to me in retrospect, but a good relearning and change commit exercise in any case!

**Comment by clarkejj** [comment]
Hi Deepesh and Marica - so once this fix is ok with all checks passing, will you then push this to develop, version-12-hotfix and other backport expected branches where required?

**Comment by deepeshgarg007** [comment]
@Mergifyio backport version-12-hotfix

**Review by deepeshgarg007 — APPROVED** [review]

**Comment by clarkejj** [comment]
Excellent thanks for the process learning coaching Deepesh and Marica

John

**Comment by marination** [comment]
@clarkejj Can you please send these changes on version-12-hotfix as well ?

**Pull request merged**